Resources/public/js/views/services/ez-locationviewviewservice.js

```diff
@@ -43,6 +44,150 @@ YUI.add('ez-locationviewviewservice', function (Y) {
+                    cancelHandler: Y.bind(function () {
+                    }, this),
+                },
+            });
+        },
+
+        /**
+         * Sends location to trash, triggering loading parent location and notifications
+         *
+         * @method _sendToTrash
+         * @protected
+         * @param {Object} e event facade of the sendToTrashAction event
+         */
+        _sendToTrash: function (e) {
+            var tasks, parentLocation, endLoadParent,
+                that = this,
+                location = this.get('location'),
+                locationId = location.get('id'),
+                contentName = e.content.get('name');
+
+            this._sendContentToTrashNotificationStarted(locationId, contentName);
+
+            tasks = new Y.Parallel();
+            endLoadParent = tasks.add();
+            this._loadParent(location, function (error, parentLocationResponse) {
+                if (error) {
+                    that._sendContentToTrashNotificationError(locationId, contentName);
+                    return;
+                }
+
+                parentLocation = parentLocationResponse.location;
+                endLoadParent();
+            });
+
+            tasks.done(function () {
+                var options = {
+                    api: that.get('capi'),
+                    parentLocation: parentLocation,
+                    contentName: contentName
+                };
+
+                location.trash(options, Y.bind(that._afterSendToTrashCallback, that, options));
+            });
+        },
+
+        /**
+         * Send to trash callback triggering notifications and making app to navigate to parent location
+         *
+         * @method _afterSendToTrashCallback
+         * @protected
+         * @param {Object} options the options for sending to trash
+         * @param {Boolean} error
+         * @param {Object} response
+         */
+        _afterSendToTrashCallback: function (options, error, response) {
+            var app = this.get('app'),
+                location = this.get('location'),
+                locationId = location.get('id'),
+                parentLocation = options.parentLocation,
+                contentName = options.contentName;
+
+            if (error) {
+                this._sendContentToTrashNotificationError(locationId, contentName);
+                return;
+            }
+
+            this._sendContentToTrashNotificationDone(locationId, contentName);
+            app.navigate(
+                app.routeUri('viewLocation', {id: parentLocation.get('id')})
+            );
+        },
+
+        /**
+         * Notification changed to *started* before sending content to trash
+         *
+         * @method _sendContentToTrashNotificationStarted
+         * @protected
+         * @param {String} locationId
+         * @param {String} contentName
+         */
+        _sendContentToTrashNotificationStarted: function (locationId, contentName) {
```

Comment:
instead of having those 3 methods that do pretty much the same, we should have a generic `_notify` method for that. @StephaneDiot is adding one in https://github.com/ezsystems/PlatformUIBundle/pull/229 so you should sync to use the same names/parameters so that when the second to merge his PR has a trivial conflict to fix.